lhs.js

```diff
@@ -15,6 +15,24 @@ function dom(selector) {
     return new DomLhs(selector);
 }
 
+/*
+ * Of all the dom nodes selected by type() or dom(), return only
+ * the fnodes that satisfy all the predicates imposed by calls to
+ * when()
+ */
+function checkPredicates(predicates, fnodes) {
+    const ret = [];
+    fnodes.forEach (function (item) {
+        const result = predicates.every (function (func) {
```

Comment:
`func` → `predicate` (or `pred` if you feel like misreads would be too likely) would make this easier to read through consistency with "predicates". And actually, this would be a good place to use an arrow function, just for brevity.